### ISSUE TITLE
No conversion between dlib pooling layer parameters and caffe pooling layer parameters found for layer 127

### ISSUE THREAD

**Issue by lunzueta** [davisking/dlib#782]
I've just tested the tool convert_dlib_nets_to_caffe included in version 19.5 with the model dlib_face_recognition_resnet_model_v1.dat and I get the error shown in the issue's title. I also can't convert to Caffe the MMOD-DNN face detector:

- Is it because of a bug somewhere?
- Is it that the conversion to Caffe of these two models isn't supported? Will they in the future?
- What kind of models can be converted to Caffe?

**Comment by davisking** [comment]
Caffe has a funny way of dealing with pooling layers.  So some pooling settings are compatible with caffe and some aren't.  It all depends on the network you are trying to convert.  You need to make sure the network architecture is something caffe can represent.  This isn't a bug.  It's just that there are things dlib does that caffe doesn't do.

As for the face and MMOD models, those use special loss layers that caffe doesn't know anything about.  So obviously those can't be converted to caffe.

**Comment by lunzueta** [comment]
OK, I see. Thanks for the clarifications. Do you consider that some day it will be possible to import-export DNN models from other frameworks, such as Caffe2, TensorFlow, PyTorch, CNTK, Chainer, etc? Currently, OpenCV 3.3 is able to import models from Caffe, TensorFlow and PyTorch. It would be great to see this possibility in dlib too (even better if there's also the possibility to export).
